input/dir/dir_input.go

```diff
@@ -0,0 +1,55 @@
+package dir
+
+import (
+	"crypto/sha512"
+	"hash"
+	"os"
+
+	"github.com/spacemonkeygo/errors"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/input"
+	"polydawn.net/repeatr/lib/fshash"
+)
+
+const Type = "dir"
+
+// interface assertion
+var _ input.Input = &Input{}
```

Comment:
Thinking out loud: maybe we want these in tests? 

Alternatively, I'm about to push a `dispatch` package that accomplish this same end elsewhere. Maybe I'll just remove these assertions in my branch.